### PR TITLE
Fix/typos advanced express tools

### PR DESCRIPTION
**Pull request by RamiroPinol**
#### Pre-Submission Checklist

- [X] Your pull request targets the `staging` branch of freeCodeCamp.
- [X] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [X] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [X] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [X] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [X] Tested changes locally.
- [ ] Closes currently open issue (replace XXXX with an issue no): Closes #XXXX

#### Description

Fixed several typos in advanced-express-tools.json

### CODE REVIEW

File: seed/challenges/06-information-security-and-quality-assurance/advanced-express-tools.json [hunk 1]
```diff
@@ -196,10 +196,10 @@
...
+        "Many strategies are set up using different settings, general it is easy to set it up based on the README in that strategies repository though. A good example of this is the Github strategy where we don't need to worry about an username or password because the user will be sent to Github's auth page to authenticate and as long as they are logged in and agree then Github returns their profile for us to use.",
```
Comment (erictleung) [inline seed/challenges/06-information-security-and-quality-assurance/advanced-express-tools.json]: `don't need to worry about an username`

This doesn't need to be changed. It should be left alone as

`don't need to worry about a username`

Source: http://english.stackexchange.com/a/105117

File: seed/challenges/06-information-security-and-quality-assurance/advanced-express-tools.json [hunk 2]
```diff
@@ -223,17 +223,17 @@
...
+        "<em>passport.authenticate</em> can also take some options as an argument such as: <code>{ failureRedirect: '/' }</code> which is incredibly useful so be sure to add that in as well. As a response after using the middleware (which will only be called if the authentication middleware passes) should be to redirect the user to <em>/profile</em> and that route should render the view 'profile.pug'.",
         "If the authentication was successful, the user object will be saved in <em>req.user</em>.",
-        "Now at this point if you enter a username and password in the form, it should redirect to the home page <em>/</em> and in the console of your server should be 'User {USERNAME} attempted to log in.' since we currently cannot login a user who isn't registered.",
+        "Now at this point if you enter an username and password in the form, it should redirect to the home page <em>/</em> and in the console of your server should be 'User {USERNAME} attempted to log in.' since we currently cannot login a user who isn't registered.",
```
Comment (erictleung) [inline seed/challenges/06-information-security-and-quality-assurance/advanced-express-tools.json]: Same here, it should be `a username`, not `an username`

File: seed/challenges/06-information-security-and-quality-assurance/advanced-express-tools.json [hunk 3]
```diff
@@ -279,9 +279,9 @@
...
-        "Also in the profile, add a link to <em>/logout</em>. That route will host the logic to unauthenticate a user. <code>a(href='/logout') Logout</code>",
+        "Now that we can ensure the user accessing the <em>/profile</em> is authenticated, we can use the information contained in 'req.user' on our page!",
+        "Go ahead and pass the object containing the variable <em>username</em> equaling 'req.user.username' into the render method of the profile view. Then go to your 'profile.pug' view and add the line <code>h2.center#welcome Welcome, #{username}!</code> creating the h2 element with the class 'center' and id 'welcome' containing the text 'Welcome, ' and the username!",
+        "Also in the profile, add a link to <em>/logout</em>. That route will host the logic to unauthenticate an user. <code>a(href='/logout') Logout</code>",
```
Comment (erictleung) [inline seed/challenges/06-information-security-and-quality-assurance/advanced-express-tools.json]: Same here, should be `unauthenticate a user` not `unauthenticate an user`

File: seed/challenges/06-information-security-and-quality-assurance/advanced-express-tools.json [hunk 4]
```diff
@@ -484,9 +484,9 @@
...
+        "With a findAndModify, it allows you to search for an object and update it, as well as upsert the object if it doesn't exist and receive the new object back each time in our callback function. In this example, we always set the last_login as now, we always increment the login_count by 1, and only when we insert a new object(new user) do we populate the majority of the fields. Something to notice also is the use of default values. Sometimes a profile returned wont have all the information filled out or it will have been chosen by the user to remain private; so in this case we have to handle it to prevent an error.",
```
Comment (erictleung) [inline seed/challenges/06-information-security-and-quality-assurance/advanced-express-tools.json]: While you're making changes, can you also add an apostrophe here?

```diff
- returned wont have all 
+ returned won't have all 
```

**Review by erictleung — CHANGES_REQUESTED** [review]
Thanks for the PR! There are some changes with having "a" in front of words start with "u". Also, there's a missing apostrophe. Let us know if you have any questions about amending your commit.

**Comment by erictleung** [comment]
@RamiroPinol also, it appears you've merged in some really old commits. You might want to head over to the [Contributors](https://gitter.im/FreeCodeCamp/Contributors) help room if you need assistance.

File: seed/challenges/06-information-security-and-quality-assurance/advanced-express-tools.json [hunk 5]
```diff
@@ -168,8 +168,8 @@
       "id": "5895f70cf9fc0f352b528e67",
       "title": "Implement the Serialization of a Passport User",
       "description": [
-        "Right now we're not loading an actually users object since we havn't set up our database. This can be done many different ways, but for our project we will connect to the database once when we start the server and keep a persistant connection for the full life-cycle of the app.",
-        "To do this, add MongoDB as a depenacy and require it in your server. (<code>const mongo = require('mongodb').MongoClient;</code>)",
+        "Right now we're not loading an actually users object since we haven't set up our database. This can be done many different ways, but for our project we will connect to the database once when we start the server and keep a persistent connection for the full life-cycle of the app.",
+        "To do this, add MongoDB as a dependancy and require it in your server. (<code>const mongo = require('mongodb').MongoClient;</code>)",
```
Comment (systimotic) [inline seed/challenges/06-information-security-and-quality-assurance/advanced-express-tools.json]: Can you change "dependancy" to "dependency"?

**Review by systimotic — CHANGES_REQUESTED** [review]
Thank you for these fixes!

File: seed/challenges/06-information-security-and-quality-assurance/advanced-express-tools.json (hunk 5 above)
Comment (RamiroPinol) [inline seed/challenges/06-information-security-and-quality-assurance/advanced-express-tools.json]: There are several "dependancy" in the json, I looked in the web and it seems to be a valid word, that's why I didn't changed it. But can fix them, no problem.

**Comment by RamiroPinol** [comment]
It seems ok now

**Comment by raisedadead** [comment]
@Greenheart @erictleung I have rebased the merge conflicts, please have a look and merge.

File: seed/challenges/06-information-security-and-quality-assurance/advanced-express-tools.json [hunk 6]
```diff
@@ -121,7 +121,7 @@
       "id": "5895f70cf9fc0f352b528e66",
       "title": "Serialization of a User Object",
       "description": [
-        "Serialization and deserialization are important concept in reguards to authentication. To serialize an object means to convert its contents into a small <em>key</em> essentially that can then be deserialized into the original object. This is what allows us to know whos communicated with the server without having to send the authentication data like username and password at each request for a new page.",
+        "Serialization and deserialization are important concept in regards to authentication. To serialize an object means to convert its contents into a small <em>key</em> essentially that can then be deserialized into the original object. This is what allows us to know whos communicated with the server without having to send the authentication data like username and password at each request for a new page.",
```
Comment (Greenheart) [inline seed/challenges/06-information-security-and-quality-assurance/advanced-express-tools.json]:
```diff
-concept
+concepts
```

**Review by Greenheart — CHANGES_REQUESTED** [review]
@RamiroPinol Nice work!

I caught one minor detail, but apart from that, this is solid :blush:

File: seed/challenges/06-information-security-and-quality-assurance/advanced-express-tools.json (hunk 6 above)
Comment (RamiroPinol) [inline seed/challenges/06-information-security-and-quality-assurance/advanced-express-tools.json]: Nice catch, fixed. ;)

**Review by Greenheart — CHANGES_REQUESTED** [review]
@RamiroPinol Great work!

Normally, you would have to squash your commits into one [using this method](https://github.com/freeCodeCamp/freeCodeCamp/blob/staging/CONTRIBUTING.md#squash-your-commits), but as I assume you haven't updated your local branch yet (`git pull origin <your-branch>`), you probably just could ignore the above and do the following instead:
1. Re-apply the changes you made through the GitHub interface to your local branch.
2. `git commit --amend --no-edit` (to add your changes to the latest commit, without changing the commit message)
3. git push origin <your-branch> -f

Then we should only see one commit here and this PR should be good to go :blush:

**Comment by RamiroPinol** [comment]
Sorry, forgot about the "only 1 commit" rule. Fixed now

**Review by Greenheart — APPROVED** [review]
@RamiroPinol No problem, you've done a great job! :smile: :+1:

File: seed/challenges/06-information-security-and-quality-assurance/advanced-express-tools.json (hunk 5 above)
Comment (systimotic) [inline seed/challenges/06-information-security-and-quality-assurance/advanced-express-tools.json]: Thanks! "dependancy" is incorrect, it's the archaic form of "dependency" https://en.wiktionary.org/wiki/dependancy
Do you want to look at replacing the other instances of this?

**Pull request merged**